impeller/renderer/backend/metal/command_buffer_mtl.mm

```diff
@@ -171,6 +176,53 @@ static bool LogMTLCommandBufferErrorIfPresent(id<MTLCommandBuffer> buffer) {
   return true;
 }
 
+bool CommandBufferMTL::SubmitCommandsAsync(
+    std::shared_ptr<RenderPass> render_pass) {
+  TRACE_EVENT0("impeller", "CommandBufferMTL::SubmitCommandsAsync");
+  if (!IsValid() || !render_pass->IsValid()) {
+    return false;
+  }
+  auto context = context_.lock();
+  if (!context) {
+    return false;
+  }
+  [buffer_ enqueue];
+  auto buffer = buffer_;
+  buffer_ = nil;
+
+  auto worker_task_runner = ContextMTL::Cast(*context).GetWorkerTaskRunner();
+
+  auto mtl_render_pass = static_cast<RenderPassMTL*>(render_pass.get());
+
+  // Render command encoder creation has been observed to exceed the stack size
+  // limit for worker threads, and therefore is intentionally constructed on the
+  // raster thread.
+  auto render_command_encoder =
+      [buffer renderCommandEncoderWithDescriptor:mtl_render_pass->desc_];
+  if (!render_command_encoder) {
+    return false;
+  }
+
+  auto task = fml::MakeCopyable(
+      [render_pass, buffer, render_command_encoder, context]() {
```

Comment:
Should we instead capture the weak_ptr to context and make sure it's still lockable in the callback here?

The advantage being that if the context has otherwise gone away, we avoid doing work.

Otherwise, I think we might need to have something about the GPU sync switch in here to make sure we're not doing encoding work when the GPU is unavailable.

Reply:
Yeah, capturing the context seems reasonable. I'm not sure what the buffer itself will do, I guess we should probably assume it won't handle anything gracefully.

Reply:
Done